core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClient.java

```diff
@@ -145,31 +106,6 @@ protected static void onRetryingComplete(ClientRequestContext ctx) {
         ctx.logBuilder().endResponseWithLastChild();
     }
 
-    /**
-     * Returns the {@link RetryRule}.
-     *
-     * @throws IllegalStateException if the {@link RetryRule} is not set
-     */
-    protected final RetryRule retryRule() {
-        checkState(retryRule != null, "retryRule is not set.");
-        return retryRule;
-    }
-
-    /**
-     * Returns the {@link RetryRuleWithContent}.
-     *
-     * @throws IllegalStateException if the {@link RetryRuleWithContent} is not set
-     */
-    protected final RetryRuleWithContent<O> retryRuleWithContent() {
-        checkState(retryRuleWithContent != null, "retryRuleWithContent is not set.");
-        return retryRuleWithContent;
-    }
```

Comment:
Armeria follows [semantic versioning](https://semver.org/). Unfortunately, we can't remove these `protected` methods in 1.x.

Reply:
Makes sense. I adjusted the code to make the client retain retryConfig if it was set directly (instead of mapping), and used it for these getters.